### ISSUE TITLE
'Client' object has no attribute 'security_groups'

### ISSUE THREAD

**Issue by akmb** [gc3-uzh-ch/elasticluster#425]
Hi,
trying to start a slurm cluster on ubuntu 16.04 (with python 2.7) fails with
```
ERROR Could not start node `frontend001`: 'Client' object has no attribute 'security_groups' -- <type 'exceptions.AttributeError'>
ERROR Could not start node `compute001`: 'Client' object has no attribute 'security_groups' -- <type 'exceptions.AttributeError'>
```
with config
```
[cloud/hobbes]
provider=openstack
auth_url=###
username=###
password=###
project_name=###
request_floating_ip=FALSE

[login/ubuntu]
image_user=ubuntu
image_user_sudo=root
image_sudo=True
user_key_name=elasticluster
user_key_private=~/.ssh/id_rsa
user_key_public=~/.ssh/id_rsa.pub

[setup/ansible-slurm]
provider=ansible
frontend_groups=slurm_master
compute_groups=slurm_worker

[cluster/slurm]
cloud=hobbes
login=ubuntu
setup=ansible-slurm
security_group=default
# Ubuntu image
image_id=###
flavor=###
frontend_nodes=1
compute_nodes=2
ssh_to=frontend
network_ids=###
```

**Comment by riccardomurri** [comment]
OpenStack must have done some other backwards-incompatible change on `novaclient`...

Can you please post the output of `pip freeze` from the virtualenv where you are running ElastiCluster?

**Comment by akmb** [comment]
this is the output of `pip freeze`
```
ansible==2.3.0.0
appdirs==1.4.3
asn1crypto==0.22.0
azure==1.0.3
azure-common==1.1.5
azure-mgmt==0.20.2
azure-mgmt-common==0.20.0
azure-mgmt-compute==0.20.1
azure-mgmt-network==0.20.1
azure-mgmt-nspkg==2.0.0
azure-mgmt-resource==0.20.1
azure-mgmt-storage==0.20.0
azure-nspkg==2.0.0
azure-servicebus==0.20.1
azure-servicemanagement-legacy==0.20.2
azure-storage==0.20.3
Babel==2.4.0
boto==2.46.1
cffi==1.10.0
click==6.7
coloredlogs==6.1
cryptography==1.8.1
debtcollector==1.13.0
-e git://github.com/gc3-uzh-ch/elasticluster.git@9e785be1bc3e5d4343b8bd2a8583bbff9204b0cd#egg=elasticluster
enum34==1.1.6
funcsigs==1.0.2
futures==3.1.1
google-api-python-client==1.6.2
google-compute-engine==2.3.5
httplib2==0.10.3
humanfriendly==2.4
idna==2.5
ipaddress==1.0.18
iso8601==0.1.11
Jinja2==2.9.6
keystoneauth1==2.19.0
MarkupSafe==1.0
monotonic==1.3
msgpack-python==0.4.8
netaddr==0.7.19
netifaces==0.10.5
oauth2client==4.0.0
oslo.i18n==3.15.0
oslo.serialization==2.18.0
oslo.utils==3.25.0
packaging==16.8
paramiko==2.1.2
pbr==3.0.0
pkg-resources==0.0.0
positional==1.1.1
prettytable==0.7.2
pyasn1==0.2.3
pyasn1-modules==0.0.8
pyCLI==2.0.3
pycparser==2.17
pycrypto==2.6.1
pyparsing==2.2.0
python-dateutil==2.6.0
python-gflags==3.1.1
python-novaclient==8.0.0
pytz==2017.2
PyYAML==3.12
requests==2.12.5
rsa==3.4.2
schema==0.6.5
simplejson==3.10.0
six==1.10.0
stevedore==1.21.0
uritemplate==3.0.0
wrapt==1.10.10
```

**Comment by riccardomurri** [comment]
Can you try downgrading `python-novaclient` to version 7.0.0 and see if this fixes it?

    pip install 'python-novaclient==7.0.0'

**Comment by akmb** [comment]
jepp, this fixes ....

thanks!!

**Comment by riccardomurri** [comment]
So I've blacklisted `python-novaclient` 8.0.0 until I have some time to dig deeper and find out what the changes in the API are...

**Comment by riccardomurri** [comment]
There is a fixed version of the code in branch "issues/#425" at http://github.com/riccardomurri/elasticluster 

I will merge it into "master" after it has received some independent testing, likely early next week.

**Comment by akmb** [comment]
This works for me. Tested with ubuntu 16.04. I was not sure about cloning correct branch: did it with `git clone -b "issues/#425" https://github.com/riccardomurri/elasticluster.git src`

`pip freeze`
```
ansible==2.3.0.0
appdirs==1.4.3
asn1crypto==0.22.0
azure==1.0.3
azure-common==1.1.5
azure-mgmt==0.20.2
azure-mgmt-common==0.20.0
azure-mgmt-compute==0.20.1
azure-mgmt-network==0.20.1
azure-mgmt-nspkg==2.0.0
azure-mgmt-resource==0.20.1
azure-mgmt-storage==0.20.0
azure-nspkg==2.0.0
azure-servicebus==0.20.1
azure-servicemanagement-legacy==0.20.2
azure-storage==0.20.3
Babel==2.3.4
boto==2.46.1
cffi==1.10.0
click==6.7
cliff==2.7.0
cmd2==0.7.0
coloredlogs==6.1
cryptography==1.8.1
debtcollector==1.13.0
-e git+https://github.com/riccardomurri/elasticluster.git@e8a318dc8c43dcd87ebd25f6c28c767f3c27d052#egg=elasticluster
enum34==1.1.6
funcsigs==1.0.2
functools32==3.2.3.post2
futures==3.1.1
google-api-python-client==1.6.2
google-compute-engine==2.3.6
httplib2==0.10.3
humanfriendly==3.1
idna==2.5
ipaddress==1.0.18
iso8601==0.1.11
Jinja2==2.9.6
jsonpatch==1.15
jsonpointer==1.10
jsonschema==2.6.0
keystoneauth1==2.20.0
MarkupSafe==1.0
monotonic==1.3
msgpack-python==0.4.8
netaddr==0.7.19
netifaces==0.10.5
oauth2client==4.1.0
os-client-config==1.27.0
osc-lib==1.6.0
oslo.config==4.1.0
oslo.i18n==3.15.0
oslo.serialization==2.18.0
oslo.utils==3.25.0
packaging==16.8
paramiko==2.1.2
pbr==3.0.0
pkg-resources==0.0.0
positional==1.1.1
prettytable==0.7.2
pyasn1==0.2.3
pyasn1-modules==0.0.8
pyCLI==2.0.3
pycparser==2.17
pycrypto==2.6.1
pyparsing==2.2.0
python-dateutil==2.6.0
python-gflags==3.1.1
python-glanceclient==2.6.0
python-keystoneclient==3.10.0
python-neutronclient==6.2.0
python-novaclient==8.0.0
pytz==2017.2
PyYAML==3.12
requests==2.14.2
requestsexceptions==1.2.0
rfc3986==1.0.0
rsa==3.4.2
schema==0.6.6
simplejson==3.10.0
six==1.10.0
stevedore==1.21.0
unicodecsv==0.14.1
uritemplate==3.0.0
warlock==1.2.0
wrapt==1.10.10
```

**Comment by riccardomurri** [comment]
After some testing also in #430 it seems that the patch is good, so merging into "master" now.

Thanks a lot for reporting and testing! It's awesome to be able to count on such prompt replies to testing calls!

**Comment by riccardomurri** [comment]
Fixed in commits 5246f199b1f8b84a143cdd3da4edbff4e2691b82 through e8a318dc8c43dcd87ebd25f6c28c76
